Adopt more smart pointers in WebCore/

cdumez opened the pull request:
#### 9e25f5119e3b28d5b60d91aefb59847295d99f1b
<pre>
Adopt more smart pointers in WebCore/
<a href="https://bugs.webkit.org/show_bug.cgi?id=263268">https://bugs.webkit.org/show_bug.cgi?id=263268</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/dom/ContainerNode.h:
(WebCore::ContainerNode::protectedRootNode const):
* Source/WebCore/dom/CurrentScriptIncrementer.h:
(WebCore::CurrentScriptIncrementer::CurrentScriptIncrementer):
(WebCore::CurrentScriptIncrementer::~CurrentScriptIncrementer):
(WebCore::CurrentScriptIncrementer::protectedDocument const):
* Source/WebCore/dom/CustomElementDefaultARIA.cpp:
(WebCore::isElementVisible):
(WebCore::CustomElementDefaultARIA::valueForAttribute const):
(WebCore::CustomElementDefaultARIA::setElementsForAttribute):
* Source/WebCore/dom/CustomElementReactionQueue.cpp:
(WebCore::CustomElementReactionQueue::tryToUpgradeElement):
(WebCore::CustomElementQueue::invokeAll):
(WebCore::CustomElementReactionQueue::enqueueElementOnAppropriateElementQueue):
(WebCore::CustomElementReactionStack::processQueue):
(WebCore::CustomElementReactionStack::takeElements):
* Source/WebCore/dom/CustomElementReactionQueue.h:
* Source/WebCore/dom/CustomElementRegistry.cpp:
(WebCore::CustomElementRegistry::document const):
(WebCore::enqueueUpgradeInShadowIncludingTreeOrder):
(WebCore::CustomElementRegistry::addElementDefinition):
(WebCore::CustomElementRegistry::get):
(WebCore::CustomElementRegistry::getName):
(WebCore::upgradeElementsInShadowIncludingDescendants):
* Source/WebCore/dom/CustomElementRegistry.h:
* Source/WebCore/dom/DatasetDOMStringMap.cpp:
(WebCore::DatasetDOMStringMap::ref):
(WebCore::DatasetDOMStringMap::deref):
(WebCore::DatasetDOMStringMap::isSupportedPropertyName const):
(WebCore::DatasetDOMStringMap::supportedPropertyNames const):
(WebCore::DatasetDOMStringMap::item const):
(WebCore::DatasetDOMStringMap::setNamedItem):
(WebCore::DatasetDOMStringMap::deleteNamedProperty):
(WebCore::DatasetDOMStringMap::protectedElement const):
* Source/WebCore/dom/DatasetDOMStringMap.h:

Canonical link: <a href="https://commits.webkit.org/269465@main">https://commits.webkit.org/269465@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ccdb4b16effd5a21fd83e4cc8f758d74053751e8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22531 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/188 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23614 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24438 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20859 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22791 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/1261 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23064 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21845 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22771 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19556 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25290 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/133 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20416 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26676 "Passed tests") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20661 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24508 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/114 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17974 "Passed tests") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/86 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5389 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/135 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/130 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/dom/CustomElementReactionQueue.h:
```diff
@@ -233,7 +233,7 @@ class CustomElementReactionStack : public CustomElementReactionDisallowedScope::
     WEBCORE_EXPORT void processQueue(JSC::JSGlobalObject*);
 
-    CustomElementQueue* m_queue { nullptr }; // Use raw pointer to avoid generating delete in the destructor.
+    std::unique_ptr<CustomElementQueue> m_queue;
```
Did we perf test this??

cdumez replied:
No, I'll do so now.

cdumez replied:
Personally I had trouble understanding why it would be slower. Surely we were destroying the queue before or we'd have a leak. How is using std::unique_ptr any lower than new/delete?

rniwa replied:
Oh, the issue was that every ~CustomElementReactionStack has to generate a call to ~CustomElementQueue in addition to calling processQueue (i.e. will have two branches instead of one).

cdumez replied:
All performance testing came back as neutral.

cdumez committed:
Adopt more smart pointers in WebCore/
https://bugs.webkit.org/show_bug.cgi?id=263268

Reviewed by Ryosuke Niwa.

* Source/WebCore/dom/ContainerNode.h:
(WebCore::ContainerNode::protectedRootNode const):
* Source/WebCore/dom/CurrentScriptIncrementer.h:
(WebCore::CurrentScriptIncrementer::CurrentScriptIncrementer):
(WebCore::CurrentScriptIncrementer::~CurrentScriptIncrementer):
(WebCore::CurrentScriptIncrementer::protectedDocument const):
* Source/WebCore/dom/CustomElementDefaultARIA.cpp:
(WebCore::isElementVisible):
(WebCore::CustomElementDefaultARIA::valueForAttribute const):
(WebCore::CustomElementDefaultARIA::setElementsForAttribute):
* Source/WebCore/dom/CustomElementReactionQueue.cpp:
(WebCore::CustomElementReactionQueue::tryToUpgradeElement):
(WebCore::CustomElementQueue::invokeAll):
(WebCore::CustomElementReactionQueue::enqueueElementOnAppropriateElementQueue):
(WebCore::CustomElementReactionStack::processQueue):
(WebCore::CustomElementReactionStack::takeElements):
* Source/WebCore/dom/CustomElementReactionQueue.h:
* Source/WebCore/dom/CustomElementRegistry.cpp:
(WebCore::CustomElementRegistry::document const):
(WebCore::enqueueUpgradeInShadowIncludingTreeOrder):
(WebCore::CustomElementRegistry::addElementDefinition):
(WebCore::CustomElementRegistry::get):
(WebCore::CustomElementRegistry::getName):
(WebCore::upgradeElementsInShadowIncludingDescendants):
* Source/WebCore/dom/CustomElementRegistry.h:
* Source/WebCore/dom/DatasetDOMStringMap.cpp:
(WebCore::DatasetDOMStringMap::ref):
(WebCore::DatasetDOMStringMap::deref):
(WebCore::DatasetDOMStringMap::isSupportedPropertyName const):
(WebCore::DatasetDOMStringMap::supportedPropertyNames const):
(WebCore::DatasetDOMStringMap::item const):
(WebCore::DatasetDOMStringMap::setNamedItem):
(WebCore::DatasetDOMStringMap::deleteNamedProperty):
(WebCore::DatasetDOMStringMap::protectedElement const):
* Source/WebCore/dom/DatasetDOMStringMap.h:

Canonical link: https://commits.webkit.org/269465@main
